pkg/sidecar/sidecar.go

```diff
@@ -53,53 +51,59 @@ func Run(runnerName string) error {
 
 	defer manager.Close()
 
-	return manager.Manage(ctx, func(ctx context.Context, instance *Instance) error {
+	return manager.Manage(globalctx, func(ctx context.Context, instance *Instance) error {
 		instance.S().Infow("managing instance", "instance", instance.Hostname)
 
 		defer func() {
+			instance.S().Infow("closing instance", "instance", instance.Hostname)
 			if err := instance.Close(); err != nil {
 				instance.S().Warnf("failed to close instance: %s", err)
 			}
 		}()
 
-		g, ctx := errgroup.WithContext(ctx)
-
 		// Network configuration loop.
-		g.Go(func() error {
-			err := instance.Network.ConfigureNetwork(ctx, &sync.NetworkConfig{
-				Network: "default",
-				Enable:  true,
-			})
-
-			if err != nil {
-				return err
-			}
-
-			// Wait for all the sidecars to enter the "network-initialized" state.
-			const netInitState = "network-initialized"
-			if _, err = instance.Writer.SignalEntry(ctx, netInitState); err != nil {
-				return fmt.Errorf("failed to signal network ready: %w", err)
-			}
-
-			instance.S().Infof("waiting for all networks to be ready")
-
-			if err := <-instance.Watcher.Barrier(
-				ctx,
-				netInitState,
-				int64(instance.RunEnv.TestInstanceCount),
-			); err != nil {
-				return fmt.Errorf("failed to wait for network ready: %w", err)
-			}
-
-			instance.S().Infof("all networks ready")
+		err := instance.Network.ConfigureNetwork(ctx, &sync.NetworkConfig{
+			Network: "default",
+			Enable:  true,
+		})
 
-			// Now let the test case tell us how to configure the network.
-			subtree := sync.NetworkSubtree(instance.Hostname)
-			networkChanges := make(chan *sync.NetworkConfig, 16)
-			if err := instance.Watcher.Subscribe(ctx, subtree, networkChanges); err != nil {
-				return fmt.Errorf("failed to subscribe to network changes: %s", err)
-			}
-			for cfg := range networkChanges {
+		if err != nil {
+			return err
+		}
+
+		// Wait for all the sidecars to enter the "network-initialized" state.
+		const netInitState = "network-initialized"
+		if _, err = instance.Writer.SignalEntry(ctx, netInitState); err != nil {
+			return fmt.Errorf("failed to signal network ready: %w", err)
+		}
+
+		instance.S().Infof("waiting for all networks to be ready")
+
+		if err := <-instance.Watcher.Barrier(
+			ctx,
+			netInitState,
+			int64(instance.RunEnv.TestInstanceCount),
+		); err != nil {
+			return fmt.Errorf("failed to wait for network ready: %w", err)
+		}
+
+		instance.S().Infof("all networks ready")
+
+		// Now let the test case tell us how to configure the network.
+		subtree := sync.NetworkSubtree(instance.Hostname)
+		networkChanges := make(chan *sync.NetworkConfig, 16)
+		if err := instance.Watcher.Subscribe(ctx, subtree, networkChanges); err != nil {
+			return fmt.Errorf("failed to subscribe to network changes: %s", err)
+		}
+		for {
+			select {
+			case <-ctx.Done():
```

Comment:
@Stebalien I guess this was not needed previously, as this whole block was inside an `errgroup`.

Reply:
Actually this appears to not be the case. Looks like we were never catching that particular cancel?